src/commands/executeFunction/eventGrid/OpenEventGridFileStep.ts

```diff
@@ -0,0 +1,85 @@
+import { AzExtFsExtra, AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import * as os from 'os';
+import * as path from "path";
+import * as vscode from 'vscode';
+import { type Progress } from "vscode";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { feedUtils } from "../../../utils/feedUtils";
+import { type ExecuteEventGridFunctionContext } from "./ExecuteEventGridFunctionContext";
+
+export class OpenEventGridFileStep extends AzureWizardExecuteStep<ExecuteEventGridFunctionContext> {
+    public priority: number;
+
+    public async execute(context: ExecuteEventGridFunctionContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined; }>): Promise<void> {
+        if (!context.eventSource || !context.selectedFileName || !context.selectedFileUrl) {
```

Comment:
We have a helper function that will throw an internal error like this automatically if it's null.

You could do something like
```
const eventSource = nonNullProp(context, 'eventSource');
const selectedFileName  = nonNullProp(context, 'selectedFileName ');
const selectedFileUrl = nonNullProp(context, 'selectedFileUrl ');
```

That way you can use these variables with confidence that they're defined and if they're not, we'll know exactly which one is throwing the error.

Reply:
Done, thanks!